text/0000-add-scratch-stack.md

```diff
@@ -0,0 +1,161 @@
+1. The scratch stack can use the Golang buildpack for its integration tests.
+   This will enable meaningful end-to-end integration tests for this stack. We
+   want to be able to build and run a test application during the stack
+   integration test, and golang is the best choice for this.
+
+## Rationale and Alternatives
+
+1. We could not create a scratch stack at all, and continue adding more
+   packages to the tiny stack. This would likely be unaceptable to developers
+   of applications that can be statically-linked, and will likely turn them
+   away from the Paketo project. We have already heard feedback that the tiny
+   stack is larger than some golang developers would like and is already a
+   reason not to adopt Paketo Buildpacks.
+1. We could keep the tiny stack as-is - not consider adding further packages to
+   it. This would satisfy current users at the expense of making the tiny stack
+   more useful. It also does not make Paketo more attractive to users who find
+   the tiny stack is already too large for their needs.
+1. We could create the new stack but not add support in the Golang buildpack.
+   This renders the stack effectively useless for Paketo consumers, as no
+   Paketo buildpacks would support it without modifying a buildpack and
+   re-packaging it.
+1. We could base the stack on Ubuntu 18.04 (Bionic) instead of (or as well as)
+   Ubuntu 22.04 (Jammy). This adds cost and complexity for the stacks
+   maintainers, and provides limited benefit to developers as the run image
+   would be almost identical. Additionally, Bionic will be end-of-life in April
+   2023 so this proposed stack would only be present for a few months.
+
+## Implementation
+
+1. Create a new Github repository at: `paketo-buildpacks/jammy-scratch-stack`
+1. Copy most of the existing contents from `paketo-buildpacks/jammy-tiny-stack` into this repository.
+1. Update metadata (e.g. stack name, Support URL, etc).
+1. Remove all packages from the `run` image package list except `tzdata` and `ca-certificates`.
+1. Ensure the integration tests statically links the golang test application.
+1. Run the automation, which will automatically publish an image.
+
+## Prior Art
+
+Some Paketo consumers already use an image with minimal packages installed in
+commercial/internal environments. Their feedback is that a scratch image with
+`tzdata` and `ca-certificates` packages is sufficient for their use-cases.
+
+Various literature (e.g.
+[this](https://blog.baeke.info/2021/03/28/distroless-or-scratch-for-go-apps/),
+and
+[this](https://chemidy.medium.com/create-the-smallest-and-secured-golang-docker-image-based-on-scratch-4752223b7324)
+describe the use of "scratch" images for statically-linked golang applications,
+with or without the `tzdata` and `ca-certificates` packages.
+
+## Unresolved Questions and Bikeshedding
+
+* Is the name "scratch" misleading, given we propose adding one or two
```

Comment:
+1 scratch is misleading I think

Reply:
Thanks for the feedback. I'm inclined to agree, but naming is hard.

I'd like to avoid increasing the scope of this RFC to include multiple language-family-specific stacks (e.g. `tiny-static`, `tiny-java`) for the reasons you point out @loewenstein .

Ideally I would like to see a name that is distinct from `tiny`. My preference is for `static` as that is widely adopted in the distroless context, and as such will be familiar to the target audience for this stack.

I'll update the RFC once we get to consensus.

Reply:
Maybe using one of the [metric prefixes](https://en.wikipedia.org/wiki/Metric_prefix#List_of_SI_prefixes) would come in handy when naming the stack. Something like `paketo-buildpacks/jammy-nano-stack` leaving room for stacks containing more or less packages.

Reply:
@modulo11 I see the value in that scheme. It offers flexibility in sizing and doesn't lock us into any particular set of packages.

However, I think I prefer the `static` name because it already has meaning with the target audience for this stack (developers of statically-linked applications looking for a minimal image to run their applications on).

Reply:
Let's go with `static`. I'll update the RFC contents, but I'll leave the filename as `scratch` until we merge the PR so as not to lose the comment history.